### PR TITLE
Add Notes Functionality to Person Class and Update JSON Serialization

### PR DESCRIPTION
**Pull request by pradyuprasad**
This PR introduces functionality to manage notes within `Person` objects, along with necessary updates to serialization and deserialization methods to include these notes. The changes follow the established patterns for JSON adaptation in the project, ensuring that notes can be stored and retrieved seamlessly in JSON format. This feature enables doctors (or users of the system) to store relevant notes for individuals. Related to #102 

### Key Changes:

1. **Create `Note` Class**:
   - Introduced a `Note` class to allow the storage of notes for each person.
   - Added necessary documentation for the class.

2. **Add Notes to `Person` Class**:
   - Added a `List<Note>` field in `Person` to manage multiple notes.
   - Implemented `addNote()` method to add notes to a `Person`.
   - Implemented `getNotes()` method to retrieve an immutable list of notes.
   - Used `List.copyOf()` to ensure the immutability of the returned notes.

3. **Add `JsonAdaptedNote` Class**:
   - Created `JsonAdaptedNote` class to handle JSON conversion of `Note` objects.
   - Implemented conversion methods between `Note` and `JsonAdaptedNote`.
   - Managed potential `IllegalValueException` for null content.
   - Used `@JsonCreator` and `@JsonValue` annotations for JSON mapping.

4. **Integrate Notes into `JsonAdaptedPerson`**:
   - Updated `JsonAdaptedPerson` to store and process notes.
   - Modified constructors and `toModelType()` method to handle note conversion.
   - Ensured that notes are serialized and deserialized consistently.

5. **Testing and Validation**:
   - Added tests for the new `Note` class and the notes functionality in `Person`.
   - Updated existing tests to reflect the new changes.
   - Adjusted sample data and expected behavior to include notes.

### CODE REVIEW

**Review by lavanyagarg112 — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/storage/JsonAdaptedNote.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    /**
+     * Converts this Jackson-friendly adapted note object into the model's {@code Note} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted note.
+     */
+    public Note toModelType() throws IllegalValueException {
+        if (content == null) {
+            throw new IllegalValueException(
+                    String.format(JsonAdaptedPerson.MISSING_FIELD_MESSAGE_FORMAT, "Note content"));
```
Comment (lavanyagarg112) [inline src/main/java/seedu/address/storage/JsonAdaptedNote.java:43]: You can add tests for this

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 2]
```diff
@@ -25,7 +27,7 @@ public class JsonAdaptedPersonTest {
     private static final String INVALID_ADDRESS = " ";
     private static final String INVALID_EMAIL = "example.com";
     private static final String INVALID_TAG = "#friend";
-    private static final String INVALID_ROLEOLE = "NOT_A_PATIENT"; // Add this
+    private static final String INVALID_ROLEOLE = "NOT_A_PATIENT";
```
Comment (lavanyagarg112) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:30]: I assume you meant INVALID_ROLE?

**Review by lavanyagarg112 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/storage/JsonAdaptedNote.java (hunk 1 above)
Comment (pradyuprasad) [reply src/main/java/seedu/address/storage/JsonAdaptedNote.java:43]: Done

File: src/main/java/seedu/address/model/person/Person.java [hunk 3]
```diff
@@ -174,6 +181,25 @@ public Set<Nric> getPatients() {
...
+        return List.copyOf(notes);
+    }
+
+    /**
+     * Adds a note to this person.
+     *
+     * @param note the note to add.
+     */
+    public void addNote(Note note) {
+        requireNonNull(note);
```
Comment (lavanyagarg112) [inline src/main/java/seedu/address/model/person/Person.java:199]: Good use of NonNull

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 4]
```diff
@@ -79,6 +84,7 @@ public JsonAdaptedPerson(Person source) {
         roles.addAll(source.getRoles().stream().map(JsonAdaptedRole::new).collect(Collectors.toList()));
         patients.addAll(source.getPatients());
         caregivers.addAll(source.getCaregivers());
+        notes.addAll(source.getNotes().stream().map(JsonAdaptedNote::new).collect(Collectors.toList()));
```
Comment (lavanyagarg112) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java:87]: Good use of streams

File: src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java [hunk 5]
```diff
@@ -25,7 +27,7 @@ public class JsonAdaptedPersonTest {
     private static final String INVALID_ADDRESS = " ";
     private static final String INVALID_EMAIL = "example.com";
     private static final String INVALID_TAG = "#friend";
-    private static final String INVALID_ROLEOLE = "NOT_A_PATIENT"; // Add this
+    private static final String INVALID_ROLE = "NOT_A_PATIENT";
```
Comment (lavanyagarg112) [inline src/test/java/seedu/address/storage/JsonAdaptedPersonTest.java:30]: Name looks correct now

**Review by lavanyagarg112 — APPROVED** [review]
LGTM now

**Pull request merged**